### ISSUE TITLE
zzztoj in Techmino

### ISSUE THREAD

**Issue by Gompyn** [26F-Studio/Techmino#27]
how to start:
- [ ] an API implementation @mm304321141

**Comment by Gompyn** [comment]
After the API is implemented, there can be some more issues to be resolved.

**Comment by Trebor-Huang** [comment]
I can try to implement / document an easy to use interface for adding bots to Techmino.

**Comment by Trebor-Huang** [comment]
As it stands, I consider `tbp` as largely incomplete. I'm going to base the protocol on the current interface for `9S` and `CC` in Techmino.

**Comment by ImpleLee** [comment]
zzz himself doesn't want to use that interface (in particular, the ones that are similar to current interface for `cc`), so I think maybe it is best to ask him first.

**Comment by ImpleLee** [comment]
zzztoj is not using the `tbp` interface...

**Comment by ImpleLee** [comment]
[this](https://gist.github.com/mm304321141/34ebe7f69fe40826caec1f2156bcf636) is what zzz was to implement, but ... he did not implement it.

**Comment by ImpleLee** [comment]
[this](https://gist.github.com/Gompyn/3a1c12bf9f2b52d80bf01c9b0e241e9c) is what I proposed for the API, but that is also not implemented.

**Comment by Trebor-Huang** [comment]
I think the current task is to sort the code out in techmino, and provide a clean and usable API with some documentation. There are no *de facto* standards for bot API's yet, and we're not rushing to create one.

**Comment by ImpleLee** [comment]
The API I am considering is just specific to Techmino, not any universal one.
(zzz may have a different idea, but my proposal is just specific to Techmino.)

**Comment by ImpleLee** [comment]
(btw I think the API should be in C... making every bot author to use Lua or to write a wrapper for every bot is really annoying.)

**Comment by Trebor-Huang** [comment]
Yeah, probably we can set the API in C and write a uniform wrapper. Or more conveniently (although the efficiency is questionable) we can use a RESTful API, which removes any language restrictions.

**Comment by ParticleG** [comment]
ZZZ bot would be integrated into the Techluster's [BotNode](https://github.com/26F-Studio/TechBotServer). It can be used in multiplayer rooms, not including local games.

**Comment by shoucandanghehe** [comment]
I think this issue needs to be updated, the dev group seems to have said that [zzz-techmino](https://github.com/26F-Studio/zzz-techmino) is ready, so what else is needed

**Comment by ParticleG** [comment]
> 

![image](https://user-images.githubusercontent.com/38580644/218250077-049a90ff-f9ef-4e41-90ea-006001c3d88b.png)
I don't think so...

**Comment by ImpleLee** [comment]
@flaribbit any news?
